sdk/keyvault/keyvault-keys/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -9,24 +9,52 @@ import { HttpHeaders } from "@azure/core-http";
 import { WebResource } from "@azure/core-http";
 import { AccessTokenCache, ExpiringAccessTokenCache } from "@azure/core-http";
 
+/**
+ * Representation of the Authentication Challenge
+ */
+export class AuthenticationChallenge {
+  constructor(public authorization: string, public scope: string) {
```

Comment:
nit: the feild should be authority not authorization

Reply:
Isn't the property coming from the service "authorization"? 🤔

Reply:
So it is 😄. Looks like it's just renamed in .NET.

https://github.com/Azure/azure-sdk-for-net/blob/9671ea4d52308897b6a7e4a32b2b86a0b53ee1b6/sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs#L239-L256

I think it's was possibly misnamed by the service originally which is why they also possibly send back "authorization_uri" in different service versions. Either way it doesn't really matter what we call the field since it's all internal. Your choice. However, you should probably add the other two possible pieces of data which could come back "scope" or "authorization_uri".

Reply:
Sounds good! I will do that 🌞 Thanks!

Reply:
I pushed the change! Now I will update the recordings to make sure things are ok

Reply:
@schaabs here's the commit: https://github.com/Azure/azure-sdk-for-js/pull/8558/commits/b923768e993b0e6175bc0172f50ba9009aa25d70